gdx/src/com/badlogic/gdx/graphics/Color.java

```diff
@@ -24,54 +24,60 @@
-	public static final Color FIREBRICK = new Color(0xb22222ff);
-
-	public static final Color RED = new Color(0xff0000ff);
-	public static final Color SCARLET = new Color(0xff341cff);
-	public static final Color CORAL = new Color(0xff7f50ff);
-	public static final Color SALMON = new Color(0xfa8072ff);
-	public static final Color PINK = new Color(0xff69b4ff);
-	public static final Color MAGENTA = new Color(1, 0, 1, 1);
-
-	public static final Color PURPLE = new Color(0xa020f0ff);
-	public static final Color VIOLET = new Color(0xee82eeff);
-	public static final Color MAROON = new Color(0xb03060ff);
-
-	/** the red, green, blue and alpha components **/
-	public float r, g, b, a;
+	public static final Color CLEAR = new ImmutableColor(0, 0, 0, 0);
+	public static final Color CLEAR_WHITE = new ImmutableColor(1, 1, 1, 0);
+
+	public static final Color BLUE = new ImmutableColor(0, 0, 1, 1);
+	public static final Color NAVY = new ImmutableColor(0, 0, 0.5f, 1);
+	public static final Color ROYAL = new ImmutableColor(0x4169e1ff);
+	public static final Color SLATE = new ImmutableColor(0x708090ff);
+	public static final Color SKY = new ImmutableColor(0x87ceebff);
+	public static final Color CYAN = new ImmutableColor(0, 1, 1, 1);
+	public static final Color TEAL = new ImmutableColor(0, 0.5f, 0.5f, 1);
+
+	public static final Color GREEN = new ImmutableColor(0x00ff00ff);
+	public static final Color CHARTREUSE = new ImmutableColor(0x7fff00ff);
+	public static final Color LIME = new ImmutableColor(0x32cd32ff);
+	public static final Color FOREST = new ImmutableColor(0x228b22ff);
+	public static final Color OLIVE = new ImmutableColor(0x6b8e23ff);
+
+	public static final Color YELLOW = new ImmutableColor(0xffff00ff);
+	public static final Color GOLD = new ImmutableColor(0xffd700ff);
+	public static final Color GOLDENROD = new ImmutableColor(0xdaa520ff);
+	public static final Color ORANGE = new ImmutableColor(0xffa500ff);
+
+	public static final Color BROWN = new ImmutableColor(0x8b4513ff);
+	public static final Color TAN = new ImmutableColor(0xd2b48cff);
+	public static final Color FIREBRICK = new ImmutableColor(0xb22222ff);
+
+	public static final Color RED = new ImmutableColor(0xff0000ff);
+	public static final Color SCARLET = new ImmutableColor(0xff341cff);
+	public static final Color CORAL = new ImmutableColor(0xff7f50ff);
+	public static final Color SALMON = new ImmutableColor(0xfa8072ff);
+	public static final Color PINK = new ImmutableColor(0xff69b4ff);
+	public static final Color MAGENTA = new ImmutableColor(1, 0, 1, 1);
+
+	public static final Color PURPLE = new ImmutableColor(0xa020f0ff);
+	public static final Color VIOLET = new ImmutableColor(0xee82eeff);
+	public static final Color MAROON = new ImmutableColor(0xb03060ff);
+
+	/** the red component **/
+	@Deprecated public float r;
+	/** the green component **/
+	@Deprecated public float g;
+	/** the blue component **/
+	@Deprecated public float b;
+	/** the alpha component **/
+	@Deprecated public float a;
```

Comment:
If a field is marked `@Deprecated`, it should also have `@deprecated` in its Javadocs, ideally with a replacement or reason why it is deprecated. From the documentation for the annotation:

```java
/**
...
 * It is strongly recommended that the reason for deprecating a program element
 * be explained in the documentation, using the {@code @deprecated}
 * javadoc tag. The documentation should also suggest and link to a
 * recommended replacement API, if applicable. A replacement API often
 * has subtly different semantics, so such issues should be discussed as
 * well.
 *
 * <p>It is recommended that a {@code since} value be provided with all newly
 * annotated program elements. Note that {@code since} cannot be mandatory,
 * as there are many existing annotations that lack this element value.
 *
 * <p>There is no defined order among annotation elements. As a matter of
 * style, the {@code since} element should be placed first.
 *
 * <p>The {@code @Deprecated} annotation should always be present if
 * the {@code @deprecated} javadoc tag is present, and vice-versa.
 */
```